### DISCUSSION TITLE
Button's 2nd click shows targetError

### DISCUSSION THREAD

**Discussion by taylorren** [bigskysoftware/htmx#1631] [Q&A] [answered]
I am trying to test HTMX. The following html runs OK with the first click. However, when the page is still on, and the `div1` is showing the API result, and I clicked the button again, the debug console shows:

`htmx:targetError`

Can someone explains to me what is this and how this should happen?

```
<!DOCTYPE html>
<html>
  <head>
    <script
      src="https://unpkg.com/htmx.org@1.9.3"
      integrity="sha384-lVb3Rd/Ca0AxaoZg5sACe8FJKF0tnUgR2Kd7ehUOG5GCcROv5uBIZsOqovBAcWua"
      crossorigin="anonymous"
    ></script>
  </head>
  <body>
    <button
      hx-get="https://api.rsywx.com/books"
      hx-trigger="click"
      hx-target="#div1"
      hx-swap="outerHTML"
    >
      Click!
    </button>

    <div id="div1">(Blank)</div>
  </body>
</html>
```

**Comment by Telroshan** [comment, marked as answer]
You specify `hx-swap="outerHTML"`, which replaces your whole div.

This means that after the request, `<div id="div1">(Blank)</div>` won't be part of the DOM anymore and got replaced by the response content of the request.
Thus, subsequent clicks on the button don't work because htmx cannot find an element with the id `div1` in the document anymore.

If you wish to keep the `div1`, you might want to use [`hx-swap="innerHTML"`](https://htmx.org/attributes/hx-swap/) instead, so that only its inner content gets swapped, and the div itself remains

**Reply by taylorren** [reply]
Thanks. My bad!
